src/EFCore.PG/Query/ExpressionTranslators/Internal/NpgsqlRangeTranslator.cs

```diff
@@ -37,53 +40,140 @@ namespace Npgsql.EntityFrameworkCore.PostgreSQL.Query.ExpressionTranslators.Inte
+            case nameof(NpgsqlRangeExtensions.IsStrictlyLeftOf) when VersionAtLeast(9, 2):
+                return new CustomBinaryExpression(e.Arguments[0], e.Arguments[1], "<<", typeof(bool));
 
-            case nameof(NpgsqlRangeExtensions.IsStrictlyRightOf):
-                return new CustomBinaryExpression(expression.Arguments[0], expression.Arguments[1], ">>", typeof(bool));
+            case nameof(NpgsqlRangeExtensions.IsStrictlyRightOf) when VersionAtLeast(9, 2):
+                return new CustomBinaryExpression(e.Arguments[0], e.Arguments[1], ">>", typeof(bool));
 
-            case nameof(NpgsqlRangeExtensions.DoesNotExtendRightOf):
-                return new CustomBinaryExpression(expression.Arguments[0], expression.Arguments[1], "&<", typeof(bool));
+            case nameof(NpgsqlRangeExtensions.DoesNotExtendRightOf) when VersionAtLeast(9, 2):
+                return new CustomBinaryExpression(e.Arguments[0], e.Arguments[1], "&<", typeof(bool));
 
-            case nameof(NpgsqlRangeExtensions.DoesNotExtendLeftOf):
-                return new CustomBinaryExpression(expression.Arguments[0], expression.Arguments[1], "&>", typeof(bool));
+            case nameof(NpgsqlRangeExtensions.DoesNotExtendLeftOf) when VersionAtLeast(9, 2):
+                return new CustomBinaryExpression(e.Arguments[0], e.Arguments[1], "&>", typeof(bool));
 
-            case nameof(NpgsqlRangeExtensions.IsAdjacentTo):
-                return new CustomBinaryExpression(expression.Arguments[0], expression.Arguments[1], "-|-", typeof(bool));
+            case nameof(NpgsqlRangeExtensions.IsAdjacentTo) when VersionAtLeast(9, 2):
+                return new CustomBinaryExpression(e.Arguments[0], e.Arguments[1], "-|-", typeof(bool));
 
-            case nameof(NpgsqlRangeExtensions.Union):
-                return new CustomBinaryExpression(expression.Arguments[0], expression.Arguments[1], "+", expression.Arguments[0].Type);
+            case nameof(NpgsqlRangeExtensions.Union) when VersionAtLeast(9, 2):
+                return new CustomBinaryExpression(e.Arguments[0], e.Arguments[1], "+", e.Arguments[0].Type);
 
-            case nameof(NpgsqlRangeExtensions.Intersect):
-                return new CustomBinaryExpression(expression.Arguments[0], expression.Arguments[1], "*", expression.Arguments[0].Type);
+            case nameof(NpgsqlRangeExtensions.Intersect) when VersionAtLeast(9, 2):
+                return new CustomBinaryExpression(e.Arguments[0], e.Arguments[1], "*", e.Arguments[0].Type);
 
             case nameof(NpgsqlRangeExtensions.Except):
-                return new CustomBinaryExpression(expression.Arguments[0], expression.Arguments[1], "-", expression.Arguments[0].Type);
+                return new CustomBinaryExpression(e.Arguments[0], e.Arguments[1], "-", e.Arguments[0].Type);
+
+            case nameof(NpgsqlRangeExtensions.Merge) when VersionAtLeast(9, 5):
```

Comment:
Same comment as above. I'd rather the user found out about the lack of `range_merge()` from a PostgreSQL error.